lib/jsdom/living/nodes/HTMLDialogElement-impl.js

```diff
@@ -1,8 +1,123 @@
 "use strict";
-
+const DOMException = require("domexception/webidl2js-wrapper");
 const HTMLElementImpl = require("./HTMLElement-impl").implementation;
+const { fireAnEvent } = require("../helpers/events");
+const { runAutoFocus, outsideUnopenedDialog } = require("../helpers/focusing");
+
+const ALREADY_OPEN = "Failed to execute 'showModal' on 'HTMLDialogElement': " +
+                    "The element already has an 'open' attribute, and therefore cannot be opened modally.";
+
+const NOT_CONNECTED = "Failed to execute 'showModal' on 'HTMLDialogElement': " +
+                      "The element is not in a Document.";
+
+class HTMLDialogElementImpl extends HTMLElementImpl {
+  constructor(globalObject, args, privateData) {
+    super(globalObject, args, privateData);
+    this._isModal = false;
+    this._returnValue = "";
+    this._previousFocusElement = null;
+  }
+
+  get open() {
+    return this.getAttributeNS(null, "open") !== null;
+  }
+
+  get returnValue() {
+    return this._returnValue;
+  }
+
+  set returnValue(returnValue) {
+    this._returnValue = returnValue.toString();
+  }
+
+  show() {
+    if (this.open) {
+      return;
+    }
+
+    this.setAttributeNS(null, "open", "");
+
+    if (this._previousFocusElement) {
+      this._previousFocusElement.focus();
+    }
+
+    this._dialogFocusSteps();
+  }
+
+  showModal() {
+    if (this.open) {
+      throw DOMException.create(this._globalObject, [ALREADY_OPEN, "InvalidStateError"]);
+    }
+
+    if (!this.isConnected) {
+      throw DOMException.create(this._globalObject, [NOT_CONNECTED, "InvalidStateError"]);
+    }
+
+    this.setAttributeNS(null, "open", "");
+    this._isModal = true;
+
+    this._ownerDocument._modalBlock(this);
+
+    if (this._previousFocusElement) {
+      this._previousFocusElement.focus();
+    }
+
+    this._dialogFocusSteps();
+  }
+
+  _dialogFocusSteps() {
+    if (this.inert) {
+      return;
+    }
+
+    if (runAutoFocus(this)) {
```

Comment:
runAutoFocus doesn't look like a good implementation of https://html.spec.whatwg.org/#autofocus-delegate